src/pages/HabitTracker/Habits.tsx

```diff
@@ -0,0 +1,224 @@
+  const [showModal, setShowModal] = useState(false);
+  const [newHabitName, setNewHabitName] = useState("");
+  const [newHabitCount, setNewHabitCount] = useState("");
+  const [editIndex, setEditIndex] = useState<number | null>(null);
+  const [originalHabitName, setOriginalHabitName] = useState("");
+  const [originalHabitCount, setOriginalHabitCount] = useState("");
+
+  // Load habits from local storage when the component mounts
+  useEffect(() => {
+    const storedHabits = localStorage.getItem("habits");
+    if (storedHabits) {
+      setHabits(JSON.parse(storedHabits));
+    }
+  }, []);
+
+  // Handler for showing the add habit modal
+  const handleAddHabitClick = () => {
+    setShowModal(true);
+    resetForm();
+  };
+
+  // Reset the form fields and edit state
+  const resetForm = () => {
+    setNewHabitName("");
+    setNewHabitCount("");
+    setEditIndex(null);
+    setOriginalHabitName("");
+    setOriginalHabitCount("");
+  };
+
+  // Handler for saving a new or edited habit
+  const handleSaveHabit = () => {
+    if (newHabitName && newHabitCount) {
+      const updatedHabits = [...habits];
+      // Update the habit if editing
+      if (editIndex !== null) {
+        updatedHabits[editIndex] = {
+          name: newHabitName,
+          count: parseInt(newHabitCount, 10),
+          index: editIndex + 1,
+        };
+      } else {
+        // Add the new habit
+        updatedHabits.push({
+          name: newHabitName,
+          count: parseInt(newHabitCount, 10),
+          index: habits.length + 1,
+        });
+      }
+      setHabits(updatedHabits);
+      localStorage.setItem("habits", JSON.stringify(updatedHabits));
+      setShowModal(false);
+      resetForm();
+    }
+  };
+
+  // Handler for closing the modal without saving
+  const handleCloseModal = () => {
+    setShowModal(false);
+    resetForm();
+  };
+
+  // Handler for deleting a habit
+  const handleDeleteHabit = (index: number) => {
+    // Filter out the deleted habit and reindex the remaining habits
+    const updatedHabits = habits
+      .filter((habit) => habit.index !== index)
+      .map((habit, idx) => ({ ...habit, index: idx + 1 }));
+    setHabits(updatedHabits);
+    localStorage.setItem("habits", JSON.stringify(updatedHabits));
+  };
+
+  // Handler for editing a habit
+  const handleEditHabit = (index: number) => {
+    const habitToEdit = habits.find((habit) => habit.index === index);
+    if (habitToEdit) {
+      setNewHabitName(habitToEdit.name);
+      setNewHabitCount(habitToEdit.count.toString());
+      setOriginalHabitName(habitToEdit.name);
+      setOriginalHabitCount(habitToEdit.count.toString());
+      setEditIndex(index - 1);
+      setShowModal(true);
+    }
+  };
+
+  // Handler for habit name input change with validation
+  const handleHabitNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    if (/^[a-zA-Z\s]*$/.test(value)) {
```

Comment:
I don't think any validation is required for the Name. The user should be able to put in any name for the habit, including numbers and symbols.